src/System.Security.Cryptography.Algorithms/tests/HashAlgorithmTest.cs

```diff
@@ -53,6 +53,100 @@ protected void VerifyRepeating(string input, int repeatCount, string output)
             }
         }
 
+        [Fact]
+        public void InvalidInput_Null()
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                Assert.Throws<ArgumentNullException>(() => hash.ComputeHash((byte[])null));
+                Assert.Throws<ArgumentNullException>(() => hash.ComputeHash(null, 0, 0));
+                Assert.Throws<NullReferenceException>(() => hash.ComputeHash((Stream)null));
+            }
+        }
+
+        [Fact]
+        public void InvalidInput_NegativeOffset()
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => hash.ComputeHash(Array.Empty<byte>(), -1, 0));
+            }
+        }
+
+        [Fact]
+        public void InvalidInput_NegativeCount()
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                Assert.Throws<ArgumentException>(() => hash.ComputeHash(Array.Empty<byte>(), 0, -1));
+            }
+        }
+
+        [Fact]
+        public void InvalidInput_TooBigOffset()
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                Assert.Throws<ArgumentException>(() => hash.ComputeHash(Array.Empty<byte>(), 1, 0));
+            }
+        }
+
+        [Fact]
+        public void InvalidInput_TooBigCount()
+        {
+            byte[] nonEmpty = new byte[53];
+
+            using (HashAlgorithm hash = Create())
+            {
+                Assert.Throws<ArgumentException>(() => hash.ComputeHash(nonEmpty, 0, nonEmpty.Length + 1));
+                Assert.Throws<ArgumentException>(() => hash.ComputeHash(nonEmpty, 1, nonEmpty.Length));
+                Assert.Throws<ArgumentException>(() => hash.ComputeHash(nonEmpty, 2, nonEmpty.Length - 1));
```

Comment:
For good measure, could you add a case with a zero-length array, a zero offset, and a positive count? (Unless you already have one elsewhere and I just missed it.)